src/gdb_packet.c

```diff
@@ -31,124 +31,222 @@
 
 #include <stdarg.h>
 
-void consume_remote_packet(char *const packet, const size_t size)
+typedef enum packet_state {
+	PACKET_IDLE,
+	PACKET_GDB_CAPTURE,
+	PACKET_GDB_ESCAPE,
+	PACKET_GDB_CHECKSUM_UPPER,
+	PACKET_GDB_CHECKSUM_LOWER,
+} packet_state_e;
+
+#if ENABLE_NOACKMODE == 1
+static bool noackmode = false;
+#else
+static const bool noackmode = false;
+#endif
+
+void gdb_set_noackmode(bool enable)
+{
+#if ENABLE_NOACKMODE == 1
+	/* 
+	 * If we were asked to disable no ack mode, and it was previously enabled,
+	 * it might mean we got a packet we determined to be the first of a new
+	 * GDB session, and as such it was not acknoledged (before GDB enabled no ack mode),
+	 * better late than never.
+	 * 
+	 * If we were asked after the connection was terminated, sending the ack will have no effect.
+	 */
+	if (!enable && noackmode) {
+		gdb_if_putchar(GDB_PACKET_ACK, 1U);
+	}
+
+	DEBUG_GDB("%s no ack mode\n", enable ? "Enable" : "Disable");
+	noackmode = enable;
+#else
+	if (enable)
+		DEBUG_ERROR("Tried to set no ack mode but it's not enabled in this build\n");
```

Comment:
Given space is at a premium, might we suggest that this is replaced by a `#if` check in `exec_q_noackmode()` which handles the packet like the unhandled packet handler would?

At least for `PC_HOSTED == 0` - this is nicer behaviour when space doesn't matter (BMDA)

Reply:
I opted to go with a build option that controls whether the feature is *advertised*, it will always be compiled in, meaning we can always manually enable in via GDB, but the default behavior depends on the build option (if advertised the feature will be active by default, otherwise only manually)

This means the feature will incur a fixed code size cost, but it's small enough to not worry about (I calculated 164B)

Reply:
That works 🙂  and it gives us options for how to use this, including in tooling using the remote protocol.